ElectronClient/gui/ConfigScreen.jsx

```diff
@@ -316,6 +347,15 @@ class ConfigScreenComponent extends React.Component {
 						style={selectStyle}
 						onChange={event => {
 							updateSettingValue(key, event.target.value);
+							// if Auto Detect option is ON theme should change according to preferred theme options
+							if (settings['themeAutoDetect']) {
+								// preferred dark theme should only work for dark system theme
+								if (key == 'preferredDarkTheme' && this.state.shouldUseDarkColors) {
+									updateSettingValue('theme', event.target.value);
+								} else if (key == 'preferredLightTheme' && !this.state.shouldUseDarkColors) {
+									updateSettingValue('theme', event.target.value);
+								}
+							}
 						}}
 					>
 						{items}
```

Comment:
Ideally there should not be any change to this ConfigScreen component. The setting model already provides a "show" property which you can use to hide/show certain settings based on the value of other settings. Also it's nice if the UI updates automatically when you click "auto detect" it's not necessary. In general we want to limit any bespoke code in here and instead enhance, if needed, models/Setting to specify whatever feature is needed.

Reply:
>Ideally there should not be any change to this ConfigScreen component.

There are multiple reasons to change this component.

- Every time we Enable/Disable `Auto Detect` option we need to detect system theme so to achieve that we need `bridge()` to call `shouldUseDarkColors()` function and `bridge()` is already imported in this component so it's become easy to implement this feature so when we Enable/Disable `Auto Detect` option the state will be change to detected system theme:
```javascript
this.setState({ shouldUseDarkColors: bridge().shouldUseDarkColors() });
```

- Now after enabling `Auto Detect` option when we click on `Apply` button `Theme` should change to `Preferred Light Theme` OR `Preferred Dark Theme`  option based on detected system theme to achieve the same I wrote this line:
```javascript
this.state.shouldUseDarkColors ? changeSetting = 'preferredDarkTheme' : changeSetting = 'preferredLightTheme';
```

- On changing `preferred light/dark theme` setting we also need to change `theme` setting and there's also one validation needed because we have two options visible to the user `Preferred Light Theme` & `Preferred Dark Theme` and only one option should change the `theme` based on detected system theme so to achieve the same I add this lines to the component. 
```javascript
if (key == 'preferredDarkTheme' && this.state.shouldUseDarkColors) {
    updateSettingValue('theme', event.target.value);
} else if (key == 'preferredLightTheme' && !this.state.shouldUseDarkColors) {
    updateSettingValue('theme', event.target.value);
}
```

I couldn't find any other way of implementing the specs without changing ConfigScreen component. If you have any other better way of achieving the same please let me know I will do accordingly

> The setting model already provides a "show" property which you can use to hide/show certain settings based on the value of other settings.

Oopss! I didn't noticed that. Thanks for the hint I'll use that property now.

> Also it's nice if the UI updates automatically when you click "auto detect" it's not necessary. In general we want to limit any bespoke code in here and instead enhance, if needed, models/Setting to specify whatever feature is needed.

I'm little bit confused here, are you saying that I should remove the immediate behavior of setting the active theme to preferred light/dark theme (without clicking on `Apply` button)
as shown in GIF ? Can you please elaborate a little more.

Reply:
> I'm little bit confused here, are you saying that I should remove the immediate behavior of setting the active theme to preferred light/dark theme (without clicking on Apply button) as shown in GIF ? Can you please elaborate a little more.

Yes we should remove this as settings are modified only when the user presses Save. If the theme changes immediately it means something in the app state has changed, and that should not happen.

Regarding your other questions, if you need to modify something in the app as a result of a setting being changed, you can put your code in BaseApplication::generalMiddleware. We can't have setting-specific code in ConfigScreen as it needs to remain generic and work for all settings (I know that there's specific code for sync settings, but ideally that should be moved out too).

Reply:
> Yes we should remove this as settings are modified only when the user presses Save. If the theme changes immediately it means something in the app state has changed, and that should not happen.

Okay fine I got it..I will remove that behavior.

>Regarding your other questions, if you need to modify something in the app as a result of a setting being changed, you can put your code in BaseApplication::generalMiddleware. We can't have setting-specific code in ConfigScreen as it needs to remain generic and work for all settings (I know that there's specific code for sync settings, but ideally that should be moved out too).

Thanks for the detailed feedback. I understand now and will move the code in ` BaseApplication::generalMiddleware`